rapid7/metasploit-framework
# Add module: Linux Priv Esc (OverlayFS copying bug) CVE-2023-0386

close #18766 

## Vulnerable Application

This exploit targets the Linux kernel bug in OverlayFS.

A flaw was found in the Linux kernel, where unauthorized access to the execution of the setuid file with capabilities
was found in the Linux kernel’s OverlayFS subsystem in how a user copies a capable file from a nosuid mount into another mount.
This uid mapping bug allows a local user to escalate their privileges on the system.

The vulnerability affects:

    * Linux kernel from (including) 5.11 up to (excluding) 5.15.91 and from (including) 5.16 Up to (excluding) 6.1.9

This module was successfully tested on:

    * Ubuntu kernel version 5.13.0-1021-oem on x64/amd64
    * Ubuntu kernel version 6.0.0-060000-generic on x64/amd64
    * Ubuntu kernel version 6.0.19-060019-generic on x64/amd64
    * Ubuntu kernel version 6.1.0-060100-generic on x64/amd64

### Install

1. Install Ubuntu version 22.04 LTS
2. (Optional) Change kernel version
```
sudo apt update
sudo apt install -y linux-image-5.13.0-1021-oem linux-headers-5.13.0-1021-oem
reboot
```
3. Install the required libraries
```
sudo apt update
sudo apt install -y gcc cmake fuse libfuse-dev libcap-dev
```

## Verification Steps

1. Make an Ubuntu
2. Create a meterpreter or shell payload and upload it to the Ubuntu target
3. Set up a handler for the payload
4. Launch the payload as a regular user on the Ubuntu target and connect the handler
5. Do: `use exploit/linux/local/cve_2023_0386_overlayfs_priv_esc`
6. Do: `run session=<session> lhost=<lhost>`
7. You should get a root
